squarelet/organizations/models/payment.py

```diff
@@ -210,6 +210,17 @@ def start(self):
             )
             self.subscription_id = stripe_subscription.id
 
+        # Trigger respective mailchimp journeys if this is the organization plan
+        if self.plan.slug in ("organization", "organization-annual"):
```

Comment:
I think we may want to check the plan for the organization level entitlement instead of the slug, so it includes relevant new plans, such as the new sunlight plans.